### PR TITLE
vimPlugins.render-markdown: init at 2024-07-07

### LINKED ISSUE #322117: Package request: vimPlugins.render-markdown-nvim
**Project description**

Plugin to improve viewing Markdown files in Neovim 

**Metadata**

* homepage URL:
* source URL: https://github.com/MeanderingProgrammer/markdown.nvim
* license: mit
* platforms: all

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by linsui**
## Description of changes

Close https://github.com/NixOS/nixpkgs/issues/322117

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Is there a reason to change the name of this plugin from `markdown-nvim` to `render-markdown` ?
Usually, we try to stick to upstream's name.

**Comment by linsui** [comment]
We already have a plugin named markdown.nvim so I use the name suggested in https://github.com/MeanderingProgrammer/markdown.nvim/?tab=readme-ov-file#lazynvim.

**Comment by GaetanLepage** [comment]
> We already have a plugin named markdown.nvim so I use the name suggested in https://github.com/MeanderingProgrammer/markdown.nvim/?tab=readme-ov-file#lazynvim.

Oh, OK.
@teto are you OK with this change ? Would you have it named differently ?

**Comment by teto** [comment]
> We already have a plugin named markdown.nvim so I use the name suggested in https://github.com/MeanderingProgrammer/markdown.nvim/?tab=readme-ov-file#lazynvim.

Thanks for the pointer. `render-markdown` makes sense and I am fine with using that for nixpkgs.
I would also suggest to the plugin author to rename his plugin to just that. Would make packaging simpler down the line (putting it on www.luarocks.org for instance). Or renaming it to nvim-markdown.

File: pkgs/applications/editors/vim/plugins/generated.nix [hunk 1]
```diff
@@ -9640,6 +9640,18 @@ final: prev:
     meta.homepage = "https://github.com/filipdutescu/renamer.nvim/";
   };
 
+  render-markdown = buildVimPlugin {
+    pname = "render-markdown";
```
Comment (GaetanLepage) [inline pkgs/applications/editors/vim/plugins/generated.nix:9644]: Shouldn't this be `markdown.nvim` however ?

Comment (linsui) [reply pkgs/applications/editors/vim/plugins/generated.nix:9644]: I thought it should match the alias. That's what samodostal-image-nvim does.

**Comment by linsui** [comment]
> I would also suggest to the plugin author to rename his plugin to just that. Would make packaging simpler down the line (putting it on [www.luarocks.org](http://www.luarocks.org/) for instance). Or renaming it to nvim-markdown

Given that the author is aware this problem maybe they just like this name...

File: pkgs/applications/editors/vim/plugins/generated.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/applications/editors/vim/plugins/generated.nix:9644]: Ok my bad then.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 325112` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vimPlugins.render-markdown</li>
  </ul>
</details>

**Review by GaetanLepage — APPROVED** [review]

**Pull request merged**